client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -156,6 +157,8 @@ abstract class AbstractTrinoResultSet
                     .add("date", String.class, java.time.LocalDate.class, string -> parseDate(string, CURRENT_TIME_ZONE, CURRENT_JAVA_TIME_ZONE).toLocalDate())
                     .add("time", String.class, Time.class, string -> parseTime(string, SYSTEM_DEFAULT_ZONE_ID))
                     .add("time with time zone", String.class, Time.class, AbstractTrinoResultSet::parseTimeWithTimeZone)
+                    .add("timestamp", String.class, Instant.class, string -> parseTimestampAsInstant(string, SYSTEM_DEFAULT_ZONE_ID))
```

Comment:
Instant represent a point in time, so it doesn't make sense for timestamp without time zone. It would be more appropriate for timestamp with time zone